src/Sarif.Viewer.VisualStudio/ErrorList/ErrorListService.cs

```diff
@@ -34,139 +37,168 @@ public class ErrorListService
 
         public static void ProcessLogFile(string filePath, string toolFormat, bool promptOnLogConversions, bool cleanErrors)
         {
-            // For now this is being done on the UI thread
-            // and is only required due to the message box being shown below.
-            // This will be addressed when https://github.com/microsoft/sarif-visualstudio-extension/issues/160
-            // is fixed.
-            ThreadHelper.ThrowIfNotOnUIThread();
+            var taskStatusCenterService = (IVsTaskStatusCenterService)Package.GetGlobalService(typeof(SVsTaskStatusCenterService));
+            var taskProgressData = new TaskProgressData
+            {
+                CanBeCanceled = false,
+                ProgressText = null,
+            };
 
-            SarifLog log = null;
+            string fileName = Path.GetFileName(filePath);
+
+            var taskHandlerOptions = new TaskHandlerOptions
+            {
+                ActionsAfterCompletion = CompletionActions.None,
+                TaskSuccessMessage = string.Format(CultureInfo.CurrentCulture, Resources.CompletedProcessingLogFileFormat, fileName),
+                Title = string.Format(CultureInfo.CurrentCulture, Resources.ProcessingLogFileFormat, fileName),
+            };
+
+            ITaskHandler taskHandler = taskStatusCenterService.PreRegister(taskHandlerOptions, taskProgressData);
 
-            string logText;
+            taskHandler.RegisterTask(ProcessLogFileAsync(filePath, toolFormat, promptOnLogConversions, cleanErrors));
+        }
+
+        public static async System.Threading.Tasks.Task ProcessLogFileAsync(string filePath, string toolFormat, bool promptOnLogConversions, bool cleanErrors)
```

Comment:
>System.Threading.Tasks. [](start = 28, length = 23)

Not necessary -- you have the `using` for it. Or... is there an ambiguity?

Reply:
It's ambiguous. 

---
In reply to: [494526058](https://github.com/microsoft/sarif-visualstudio-extension/pull/202#discussion_r494526058) [](ancestors = 494526058)